string.c

```diff
@@ -3347,6 +3347,63 @@ rb_str_cat_cstr(VALUE str, const char *ptr)
     return rb_str_buf_cat(str, ptr, strlen(ptr));
 }
 
+static void
+rb_str_buf_cat_byte(VALUE str, char byte)
```

Comment:
"Plain" `char` may equal to `signed` or `unsigned`.
Should not use it in the case it may not be 7bit clean.
In this function, the argument to `ISASCII` may be promoted to a negative `int`, for instance.

```suggestion
rb_str_buf_cat_byte(VALUE str, unsigned char byte)
```

Reply:
Thank you for the explanation. I'll be sure to use `unsigned char` in future work.